pkg/prommetrics/prommetrics.go

```diff
@@ -39,7 +39,7 @@ const (
 )
 
 var (
-	metricLabels = []string{"namespace", "metric", "scaledObject", "scaler", "scalerIndex"}
+	metricLabels = []string{"namespace", "metric", "scaledObject", "scaler", "scalerIndex", "type"}
```

Comment:
Considering that "scaledObject" may refer to a scaledJob instead, would it be better to use "scaledResourceName" or another term instead?

Reply:
Thanks for the feedback.
BTW,  I think it could be breaking change current label. @JorTurFer How do you think?

Reply:
Yeah, sadly we should add the second label end deprecate the other following the deprecation policy (announce, 2 versions until removal, etc)

Reply:
But even that, I think it's a good improvement if we unify them under the single labels. (following the deprecation policy, ofc)
WDYT @zroubalik @tomkerkhove ?

Reply:
Hi, @JorTurFer 
Does that mean we can move forward with my current change?
I will update this PR again, if my understanding is wrong.

Reply:
If this is a breaking change, then we cannot - https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md

We need to introduce new next to old approach

Reply:
@tomkerkhove @JorTurFer 
How about adding a new label like this one and gradually change it next time?
```suggestion
	metricLabels = []string{"namespace", "metric", "scaledObject", "scaledJob", "scaler", "scalerIndex"}
```

or 
```suggestion
	metricLabels = []string{"namespace", "metric", "scaledObject", "scaledJob", "scaler", "scalerIndex", "type"}
```

Reply:
Adding more labels isn't a breaking change IMO because it still works, but we can rethink the metrics to unify them if you think that it's better. 
AFAIK, having the same metric for different resources exposed split isn't correct from prometheus pov. I mean, something like `scaledobject_error` and `scaledjob_error` should be `errors` with the type as a label. This is because prometheus allows to filter them easily and if we expose them split, users have to aggregate multiple metrics them from their side (making the query more complex), which I'd say it's against the working way in prometheus. It the same as exposing errors by resource and total errors, prometheus can aggregate them just with a sum. @pragmaticivan WDYT?

Reply:
If you have any behavior in the future that will indeed be different between scaledjob and a scaled object, having a new metric is totally fine. 

The more labels with multiple values you add to a metric, the more cardinality you are going to have.

So maybe if you're going to add the `job/object` name somehow as a value, it might actually be cheaper/efficient to split into 2 metrics because if you have a cluster with far too many resources, you are multiplying the number of possible names vs number of possible types.

Reply:
I do see that it might be too late to consider splitting things though  
https://github.com/kedacore/keda/blob/84c81dc346c8382bbde55339695a68c5a4b732b1/pkg/prommetrics/prommetrics.go#L69-L95

Reply:
Here's my interpretation of how this all should look like https://docs.google.com/spreadsheets/d/1arVI8T_pdUO5ObTUrNtmnIfm07rK0PY8dSwPLvcmsW0/edit#gid=49546810

(Second tab)

There might be a better naming for `scalerType` but the current ones are not clear enough for someone operating this for the first time.

Reply:
Hi All. Thanks for bringing this up. The discussion means so much to me :) Due to my vacation, I couldn't fully focus on it. 

BTW, @pragmaticivan. Are you suggesting the change like this way?
```suggestion
	metricLabels = []string{"namespace", "metric", "scaledObject", "scaler", "scalerIndex", "scalerType"}
```
Once you confirmed or suggest a better way, I will updated it. Thanks!

Reply:
Hi Any update?

Reply:
I would actually prefer, if we have a separate scaledObject and scaledJob type labels.

Reply:
I also prefer to have a separate scaledObject and scaledJob type labels too as it's more clear.

Reply:
@zroubalik @SpiritZhou Thanks for the feedback :) It means so much to me.
Then I need to move back to my previous change. Similar thing is here. (It's on the different PR, cause this PR was broken)
https://github.com/kedacore/keda/pull/4817/commits/fae57e6b852c6446d02d406d82c4f4d56db791f0

BTW, I think @JorTurFer suggests to use same label cause it increase code complexity.
How do you think? you left comment here.
https://github.com/kedacore/keda/pull/4817
<img width="923" alt="image" src="https://github.com/kedacore/keda/assets/38376255/c497d775-ab39-4b4b-8e47-4a66e070c52b">

BTW, I prefer to take current approach,  cause it seems more simple for me :) But i don't mind any choice.

Reply:
BTW, once decide, I will move forward. Thanks everyone for these nice review.

Reply:
Hi. everyone. I will take the separate label approach. Cause I rolled it back twice after reviews, I have been waiting the review again. 
Thanks for the all support.
I will update the code within this week.

Reply:
Thanks a lot for the patience 🙏